PowerShellTools/DebugEngine/BreakpointManager.cs

```diff
@@ -0,0 +1,240 @@
+                    {
+                        EnableBreakpoint(bp, 0);  // Disable PS breakpoint
+                    }
+                }
+            }
+        }
+        
+        /// <summary>
+        /// Placeholder for future support on debugging command in REPL window
+        /// Breakpoint has been updated
+        /// </summary>
+        /// <param name="e"></param>
+        public void UpdateBreakpoint(DebuggerBreakpointUpdatedEventArgs e)
+        {
+            Log.InfoFormat("Breakpoint updated: {0} {1}", e.UpdateType, e.Breakpoint);
+
+            if (BreakpointUpdated != null)
+            {
+                BreakpointUpdated(this, e);
+            }
+        }
+
+        /// <summary>
+        /// Process line breakpoint when break
+        /// </summary>
+        /// <param name="script">Script file</param>
+        /// <param name="line">Line of breakpoint</param>
+        /// <param name="column">Column of breakpoint</param>
+        /// <returns>Success or not</returns>
+        public bool ProcessLineBreakpoints(string script, int line, int column)
+        {
+            Log.InfoFormat("Process Line Breapoints");
+
+            var bp =
+                _breakpoints.FirstOrDefault(
+                    m =>
+                    m.Column == column && line == m.Line &&
+                    script.Equals(m.File, StringComparison.InvariantCultureIgnoreCase));
+
+            if (bp != null)
+            {
+                if (BreakpointHit != null)
+                {
+                    Log.InfoFormat("Breakpoint @ {0} {1} {2} was hit.", bp.File, bp.Line, bp.Column);
+                    BreakpointHit(this, new EventArgs<ScriptBreakpoint>(bp));
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Add breakpoint
+        /// </summary>
+        /// <param name="breakpoint">Breakpoint to be added</param>
+        public void SetBreakpoint(ScriptBreakpoint breakpoint)
+        {
+            Log.InfoFormat("SetBreakpoint: {0} {1} {2}", breakpoint.File, breakpoint.Line, breakpoint.Column);
+
+            try
+            {
+                if (Debugger.DebuggingService.GetRunspaceAvailability() == RunspaceAvailability.Available)
+                {
+                    Debugger.DebuggingService.SetBreakpoint(new PowershellBreakpoint(breakpoint.File, breakpoint.Line, breakpoint.Column));
+                }
+                else
+                {
+                    Debugger.ExecuteDebuggingCommand(string.Format("Set-PSBreakpoint -Script \"{0}\" -Line {1}", breakpoint.File, breakpoint.Line));
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error("Failed to set breakpoint.", ex);
+            }
+        }
+
+        /// <summary>
+        /// Enable breakpoint
+        /// </summary>
+        /// <param name="breakpoint">Breakpoint to be added</param>

```

Comment:
comment doesn't include fEnable.  [Did we want to have EnableBreakPoint( breakpoint, bool enable ) as well]?